tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -455,7 +455,7 @@ where
         let mut transaction_errors = vec![None; transactions.len()];
         let mut confirmed_transactions = 0;
         let mut block_height = self.rpc_client.get_block_height().await?;
-        while expired_blockhash_retries > 0 {
+        while expired_blockhash_retries >= 0 {
```

Comment:
If you're only looking to fix the print out, why not move the decrement before the print out? Using `>=` means the loop could execute 6 times now (instead of existing behavior of 5).

For what it is worth, I could see the variable name being interpreted as an initial attempt + 5 more additional attempts for a total of 6. But, not sure that we want to allow an extra attempt here.

Reply:
From the wording and the message, the original code seemed specifically to say **re**tries instead of tries, so that's why I changed it this way. If you prefer, I can update the message and the variable counter to say "tries" instead.

Reply:
@steviez I updated the counter to `4` instead of `5` to match the previous behavior.